src/FlatSharp/TypeModel/Vectors/UnityNativeArrayVectorTypeModel.cs

```diff
@@ -0,0 +1,57 @@
+namespace FlatSharp.TypeModel.Vectors;
+
+/// <summary>
+/// Defines a vector type model for a Unity NativeArray collection.
+/// </summary>
+public class UnityNativeArrayVectorTypeModel : BaseVectorTypeModel
+{
+    internal UnityNativeArrayVectorTypeModel(Type vectorType, TypeModelContainer provider) : base(vectorType, provider)
+    {
+    }
+
+    public override string LengthPropertyName => nameof(Memory<byte>.Length);
+
+    public override Type OnInitialize()
+    {
+        return this.ClrType.GetGenericArguments()[0];
+    }
+
+    protected override string CreateLoop(FlatBufferSerializerOptions options, string vectorVariableName, string numberOfItemsVariableName, string expectedVariableName, string body)
+    {
+        FlatSharpInternal.Assert(false, "Not expecting to do loop get max size for memory vector");
+        throw new Exception();
+    }
+
+    public override CodeGeneratedMethod CreateParseMethodBody(ParserCodeGenContext context)
+    {
+        ValidateWriteThrough(
+            writeThroughSupported: false,
+            this,
+            context.AllFieldContexts,
+            context.Options);
+
+        var lazySuffix = context.Options.Lazy ? "Lazy" : "";
+
+        var readNativeArray = $"{context.InputBufferVariableName}.ReadNativeArrayBlock{lazySuffix}<{this.ItemTypeModel.GetGlobalCompilableTypeName()}, {context.InputBufferTypeName}>({context.OffsetVariableName})";
+
+        return new CodeGeneratedMethod($"return {readNativeArray};");
+    }
+
+    public override CodeGeneratedMethod CreateSerializeMethodBody(SerializationCodeGenContext context)
+    {
+        var writeNativeArray = $"{context.SpanWriterVariableName}.WriteNativeArrayBlock({context.SpanVariableName}, {context.ValueVariableName}, {context.OffsetVariableName}, " +
+                               $"{ItemTypeModel.PhysicalLayout[0].Alignment}, {this.PaddedMemberInlineSize}, {context.SerializationContextVariableName});";
```

Comment:
Let's add a validation that PhyiscalLayout is of length 1 for these. PhysicalLayout really means "vtable layout". Unions in Flatbuffers are done... strangely and are the only types that have a physical layout of length, so this really means that you can't store unions in a NativeArray. Since FlatSharp unions are structs, someone could actually get the wrong idea here :)

Reply:
Sorry, to expand on this, validations are usually done in the `Initialize` method, and failures usually throw a `InvalidFlatBufferDefinitionException` with some human-readable message.

Validations I think you need here are...:
- Value type (unless the generic definition prevents this).
- Not a union (physical layout length == 1)
- Item is a valid vector member (ItemTypeModel.IsValidVectorMember)